lib/abide_dev_utils/xccdf.rb

```diff
@@ -288,36 +287,46 @@ def diff_controls(other)
 
       def map_indexed(index: 'title', framework: 'cis', key_prefix: '')
         all_indexes = ['title', 'hiera_title', 'hiera_title_num', 'number']
-        c_map = profiles.each_with_object({}) do |profile, obj|
-          controls_hash = profile.controls.each_with_object({}) do |ctrl, hsh|
-            real_index = if index == 'hiera_title_num'
-                           ctrl.hiera_title(number_format: true)
-                         elsif index == 'title'
-                           resolve_control_reference(ctrl).xpath('./xccdf:title').text
-                         else
-                           ctrl.send(index.to_sym)
-                         end
-             controls_array = all_indexes.each_with_object([]) do |idx_sym, ary|
-              next if idx_sym == index
-    
-              item = if idx_sym == 'hiera_title_num'
-                       ctrl.hiera_title(number_format: true)
-                     elsif idx_sym == 'title'
-                       resolve_control_reference(ctrl).xpath('./xccdf:title').text
-                     else
-                       ctrl.send(idx_sym.to_sym)
-                     end
-              ary << "#{item}"
+
+        objserver = {}
+        objworkstation = {}
+
+        ['workstation', 'server'].each do |t|
```

Comment:
Instead of hardcoding this, when you loop through `profiles` it yields a `Profile` object. You can get the profile name from the object itself. Take a look at https://github.com/puppetlabs/abide_dev_utils/blob/eaf247b638825afe32428301ca9ab8fad1cac060/lib/abide_dev_utils/xccdf.rb#L635